Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -4096,6 +4127,35 @@ JSC_DEFINE_JIT_OPERATION(operationArrayIndexOfValueInt32OrContiguous, UCPUStrict
     OPERATION_RETURN(scope, toUCPUStrictInt32(-1));
 }
 
+JSC_DEFINE_JIT_OPERATION(operationArrayIndexOfValueInt32, UCPUStrictInt32, (JSGlobalObject* globalObject, Butterfly* butterfly, EncodedJSValue encodedValue, int32_t index))
+{
+    VM& vm = globalObject->vm();
+    CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
+    JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue searchElement = JSValue::decode(encodedValue);
+
+    if (!searchElement.isNumber())
+        OPERATION_RETURN(scope, toUCPUStrictInt32(-1));
+
+    int32_t length = butterfly->publicLength();
+    auto data = butterfly->contiguous().data();
+
+    if (index >= length)
+        OPERATION_RETURN(scope, toUCPUStrictInt32(-1));
+
+    for (; index < length; ++index) {
+        JSValue value = data[index].get();
+        if (!value || !value.isNumber())
+            continue;
+        if (searchElement.asInt32() == value.asNumber())
```

Comment:
`searchElement` is not guaranteed as `Int32`. So you cannot use `asInt32`.
You should check whether the contained content is Int32 via `isInt32AsAnyInt` and `asInt32AsAnyInt`.
On the other hand, value is ensured as int32, so you should use `asInt32()`.

Reply:
Fixed!